lucene/queries/src/java/org/apache/lucene/queries/intervals/MultiTermIntervalsSource.java

```diff
@@ -96,7 +96,7 @@ public MatchesIterator matches(String field, LeafReaderContext ctx, int doc) thr
 
   @Override
   public void visit(String field, QueryVisitor visitor) {
-
+    visitor.visitLeaf(new IntervalQuery(field, this));
```

Comment:
Isn't it better to stub AtomatonQuery and yield it here. IIRC it resolves simplest MTQ highlighting cases as nobrainer.

Reply:
I don't really like the idea of reporting back an entirely different query here.  I think for highlighting multi-terms we need to extend QueryVisitor to add a `consumeTermsMatching(CompiledAutomaton ca)` method, or similar.  I'll open a follow-up to discuss - there are tricky things involved with char-vs-byte automatons etc.